File: Software Development Module/Week9/AverageApp.java
/*
AverageApp.java
Billy Cussen
18/07/2020
*/

import java.util.Scanner;
import java.util.InputMismatchException;

public class AverageApp {
    
    public static void main(String [] args){

        //Variables
        int[] input;
        int arraySize = 0;
        String output;
        boolean inputCheckAverage;

        //Objects
        Scanner keyboard;
        Average myAverage;

        //Input
        do{
            inputCheckAverage = true;
            keyboard = new Scanner(System.in);

            try{
                System.out.print("Please enter the size of your Array: ");
                arraySize = keyboard.nextInt();

                if(arraySize <= 2 || arraySize > 10){
                    throw new InputMismatchException();
                }

            }catch(InputMismatchException e){
                System.out.println("You need to enter an integer greater than 2 and less than 11");
                inputCheckAverage = false;
            }finally{
                keyboard = null;
            }

        }while(!inputCheckAverage);

        input = new int[arraySize];

        for(int i = 0; i < input.length; i++){

            do{
                inputCheckAverage = true;
                keyboard = new Scanner(System.in);

                try{
                    System.out.print("Please enter number "+(i+1)+" of your Array: ");
                    input[i] = keyboard.nextInt();

                    if(input[i] <= 0){
                        throw new InputMismatchException();
                    }

                }catch(InputMismatchException f){
                    System.out.println("You need to enter an Integer that is also greater than 0");
                    inputCheckAverage = false;
                } finally{
                    keyboard = null;
                }

            }while(!inputCheckAverage);

        }

        myAverage = new Average(input);

        //Process & Output
        output = myAverage.computeAverage();
        myAverage = null;
        System.out.println(output);
    }

}

File: Python Course/Software Development Module - in Python/Week9/AverageApp.py
"""
AverageApp.py
Billy Cussen
01/10/2020
"""

#Object
import gc
import Average
from Average import myAverage

#Variables
repeat = "y"

while repeat == "y":

    inputCheck = True

    while inputCheck:
        try:
            arraySize = int(input("Please enter the Size of the Array (between 1 and 10): "))
        except:
            print("You need to enter an Integer between 1 and 10!")
        else:
            if arraySize <= 0 or arraySize > 10:
                print("You need to enter an Integer between 1 and 10!")
            else:
                inputCheck = False
    
    arrayNumbers = []

    for i in range(arraySize):
        inputCheck = True
        while inputCheck:
            try:
                arrayNumbers.append(int(input("Please enter Number "+str(i+1)+" of the Array: ")))
            except:
                print("You need to enter an Integer!")
            else:
                if arrayNumbers[i] <= 0:
                    print("You need to enter an Integer greater than 0!")
                else:
                    inputCheck = False

    #Process & Output
    myAverageApp = myAverage(arrayNumbers)
    myAverageApp.calculateAverage()
    print("Average of the Array is: "+str(myAverageApp.average))
    print("Maximum Value is: "+str(myAverageApp.max))
    print("Minimum Value is: "+str(myAverageApp.min))
    myAverageApp = None
    gc.collect()

    repeat = ""
    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter \"Y\" to retry, enter \"N\" to Exit: "))
        repeat = repeat.lower()

        if repeat !="y" and repeat!="n":
            print("You need to enter \"Y\" or \"N\"")
    
print ("Thanks for using my App!")